fundrebalancer/fundrebalancer.go

```diff
@@ -263,13 +263,23 @@ func (r *FundRebalancer) MoveFundsToChain(
 		}
 		txn := txns[0]
 
+		approvalHash, err := r.ERC20Approval(ctx, txn)
+		if err != nil {
+			return nil, nil, fmt.Errorf("approving usdc erc20 spend on chain %s for %suusdc: %w", rebalanceFromChainID, usdcToRebalance.String(), err)
+		}
+
+		txnWithMetadata, err := r.TxnWithMetadata(ctx, rebalanceFromChainID, rebalanceFromChainID, usdcToRebalance, txn)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting transaction metadata: %w", err)
+		}
+
 		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(rebalanceFromChainID)
 		if err != nil {
 			return nil, nil, fmt.Errorf("getting chain config for gas threshold check: %w", err)
 		}
 
 		if chainConfig.MaxRebalancingGasThreshold != 0 {
```

Comment:
if we submit the erc20 transaction and then the rebalance fails to satisfy the threshold, then we end up not inserting the erc20 transaction into the submitted_txs table

Reply:
yeah this was originally done because i wanted to link the erc20 approve txns with the fund rebalance txns. so i needed the id from the row that is inserted for the rebalance tx at the end of the function. I got around this now by inserting a blank (no tx hash no amount) fund rebalance txn at the start of the function, and then updating it to later when the actual rebalance happens, so that the rebalance id is available to the erc20 approve tx as well (so they're all linked together now)